Kata-env and Kata-check are not architecture-independent

# Description of problem
For now, the functionality of kata-check and kata-env are missing in Arm architecture. Error outputs are as follows:

```
# github.com/kata-containers/runtime/cli
./kata-check.go:250:65: undefined: archKernelParamHandler
./kata-check.go:271:27: undefined: archRequiredCPUFlags
./kata-check.go:272:27: undefined: archRequiredCPUAttribs
./kata-check.go:273:27: undefined: archRequiredKernelModules
./kata-check.go:285:10: undefined: archHostCanCreateVMContainer
./kata-env.go:179:26: undefined: archRequiredCPUFlags
./kata-env.go:180:26: undefined: archRequiredCPUAttribs
./kata-env.go:181:26: undefined: archRequiredKernelModules
Makefile:310: recipe for target '/home/Wei/penny/go/src/github.com/kata-containers/runtime/kata-runtime' failed
make: *** [/home/Wei/penny/go/src/github.com/kata-containers/runtime/kata-runtime] Error 2

```
It could come from following aspects:
1. Struct `vmContainerCapableDetails` was architecture-relevant, for example, there were no tag showing virtualization capability in /proc/cpuinfo in aarch64, so variable `requiredCPUFlags` could be no use. `flag` filed in /proc/cpuinfo may only amd64-specific.
3. Struct `CPUInfo` was also architecture-relevant, e.g. variable vendor and model are meaningless for arm architecture. Because func `getCPUDetails()` is related to `CPUInfo`, it may not be included in universal file cli/utils.go 
4. Func `hostIsVMContainerCapable()` is only suitable for amd64 to check host's capability of creating a VM container

We think that we could move the architecture-relevant variable and func to files with specific architecture suffixes. e.g. func `getCPUFlags`, `checkCPUFlags`, `hostIsVMContainerCapable` and const `cpuFlagsTag` should be transferred to file `kata-check_amd64.go`. 

@bergwolf @gnawux @jodh-intel @weichen81

--- a/cli/kata-check.go
+++ b/cli/kata-check.go
@@ -45,11 +45,15 @@ type vmContainerCapableDetails struct {
 
 const (
 	moduleParamDir        = "parameters"
-	cpuFlagsTag           = "flags"
 	successMessageCapable = "System is capable of running " + project
 	successMessageCreate  = "System can currently create " + project
 	failMessage           = "System is not capable of running " + project
 	kernelPropertyCorrect = "Kernel property value correct"
+
+	// these refer to fields in the procCPUINFO file
+	genericCPUFlagsTag    = "flags"
+	genericCPUVendorField = "vendor_id"
+	genericCPUModelField  = "model name"
 )
 
 // variables rather than consts to allow tests to modify them

--- a/cli/kata-check_amd64.go
+++ b/cli/kata-check_amd64.go
@@ -9,6 +9,12 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	cpuFlagsTag        = genericCPUFlagsTag
+	archCPUVendorField = genericCPUVendorField
+	archCPUModelField  = genericCPUModelField
+)
+
 // archRequiredCPUFlags maps a CPU flag value to search for and a
 // human-readable description of that value.
 var archRequiredCPUFlags = map[string]string{

--- a/cli/kata-check_arm64.go
+++ b/cli/kata-check_arm64.go
@@ -5,6 +5,37 @@
 
 package main
 
+import "github.com/sirupsen/logrus"
+
+const (
+	cpuFlagsTag        = "Features"
+	archCPUVendorField = "CPU implementer"
+	archCPUModelField  = "CPU variant"
+)
+
+// archRequiredCPUFlags maps a CPU flag value to search for and a
+// human-readable description of that value.
+var archRequiredCPUFlags = map[string]string{}
+
+// archRequiredCPUAttribs maps a CPU (non-CPU flag) attribute value to search for
+// and a human-readable description of that value.
+var archRequiredCPUAttribs = map[string]string{}
+
+// archRequiredKernelModules maps a required module name to a human-readable
+// description of the modules functionality and an optional list of
+// required module parameters.
+var archRequiredKernelModules = map[string]kernelModule{
+	"kvm": {
+		desc: "Kernel-based Virtual Machine",
+	},
+	"vhost": {
+		desc: "Host kernel accelerator for virtio",
+	},
+	"vhost_net": {
+		desc: "Host kernel accelerator for virtio network",
+	},
+}
+
 // kvmIsUsable determines if it will be possible to create a full virtual machine
 // by creating a minimal VM and then deleting it.
 func kvmIsUsable() error {
@@ -19,4 +50,8 @@ func archHostCanCreateVMContainer() error {
 // of creating a VM container.
 func hostIsVMContainerCapable(details vmContainerCapableDetails) error {
 	return genericHostIsVMContainerCapable(details)
+}
+
+func archKernelParamHandler(onVMM bool, fields logrus.Fields, msg string) bool {
+	return genericArchKernelParamHandler(onVMM, fields, msg)
 }

--- a/cli/kata-check_ppc64le.go
+++ b/cli/kata-check_ppc64le.go
@@ -7,7 +7,14 @@ package main
 
 import (
 	"fmt"
+
 	"github.com/sirupsen/logrus"
+)
+
+const (
+	cpuFlagsTag        = genericCPUFlagsTag
+	archCPUVendorField = genericCPUVendorField
+	archCPUModelField  = genericCPUModelField
 )
 
 // archRequiredCPUFlags maps a CPU flag value to search for and a
@@ -37,11 +44,6 @@ func archHostCanCreateVMContainer() error {
 // hostIsVMContainerCapable checks to see if the host is theoretically capable
 // of creating a VM container.
 func hostIsVMContainerCapable(details vmContainerCapableDetails) error {
-	_, err := getCPUInfo(details.cpuInfoFile)
-	if err != nil {
-		return err
-	}
-
 	count, err := checkKernelModules(details.requiredKernelModules, archKernelParamHandler)
 	if err != nil {
 		return err

--- a/cli/kata-check_test.go
+++ b/cli/kata-check_test.go
@@ -229,10 +229,23 @@ func TestCheckGetCPUFlags(t *testing.T) {
 		{"foo", ""},
 		{"foo bar", ""},
 		{":", ""},
-		{"flags", ""},
-		{"flags:", ""},
-		{"flags: a b c", "a b c"},
-		{"flags: a b c foo bar d", "a b c foo bar d"},
+
+		{
+			cpuFlagsTag,
+			"",
+		},
+		{
+			cpuFlagsTag + ":",
+			"",
+		},
+		{
+			fmt.Sprintf("%s: a b c", cpuFlagsTag),
+			"a b c",
+		},
+		{
+			fmt.Sprintf("%s: a b c foo bar d", cpuFlagsTag),
+			"a b c foo bar d",
+		},
 	}
 
 	for _, d := range data {

--- a/cli/kata-env_test.go
+++ b/cli/kata-env_test.go
@@ -205,9 +205,13 @@ VERSION_ID="%s"
 `, expectedDistro.Name, expectedDistro.Version)
 
 	procCPUInfoContents := fmt.Sprintf(`
-vendor_id	: %s
-model name	: %s
-`, expectedCPU.Vendor, expectedCPU.Model)
+%s	: %s
+%s	: %s
+`,
+		archCPUVendorField,
+		expectedCPU.Vendor,
+		archCPUModelField,
+		expectedCPU.Model)
 
 	data := []filesToCreate{
 		{procVersion, procVersionContents},

--- a/cli/utils.go
+++ b/cli/utils.go
@@ -108,24 +108,35 @@ func getCPUDetails() (vendor, model string, err error) {
 	lines := strings.Split(cpuinfo, "\n")
 
 	for _, line := range lines {
-		if strings.HasPrefix(line, "vendor_id") {
-			fields := strings.Split(line, ":")
-			if len(fields) > 1 {
-				vendor = strings.TrimSpace(fields[1])
+		if archCPUVendorField != "" {
+			if strings.HasPrefix(line, archCPUVendorField) {
+				fields := strings.Split(line, ":")
+				if len(fields) > 1 {
+					vendor = strings.TrimSpace(fields[1])
+				}
 			}
-		} else if strings.HasPrefix(line, "model name") {
-			fields := strings.Split(line, ":")
-			if len(fields) > 1 {
-				model = strings.TrimSpace(fields[1])
+		}
+
+		if archCPUModelField != "" {
+			if strings.HasPrefix(line, archCPUModelField) {
+				fields := strings.Split(line, ":")
+				if len(fields) > 1 {
+					model = strings.TrimSpace(fields[1])
+				}
 			}
 		}
 	}
 
-	if vendor != "" && model != "" {
-		return vendor, model, nil
+	if vendor == "" {
+		return "", "", fmt.Errorf("cannot find vendor field in file %v", procCPUInfo)
 	}
 
-	return "", "", fmt.Errorf("failed to find expected fields in file %v", procCPUInfo)
+	// model name is optional
+	if archCPUModelField != "" && model == "" {
+		return "", "", fmt.Errorf("cannot find model field in file %v", procCPUInfo)
+	}
+
+	return vendor, model, nil
 }
 
 // resolvePath returns the fully resolved and expanded value of the

--- a/cli/utils_test.go
+++ b/cli/utils_test.go
@@ -225,10 +225,10 @@ func TestGetCPUDetails(t *testing.T) {
 	}
 
 	const validVendorName = "a vendor"
-	validVendor := fmt.Sprintf(`vendor_id	: %s`, validVendorName)
+	validVendor := fmt.Sprintf(`%s	: %s`, archCPUVendorField, validVendorName)
 
 	const validModelName = "some CPU model"
-	validModel := fmt.Sprintf(`model name	: %s`, validModelName)
+	validModel := fmt.Sprintf(`%s	: %s`, archCPUModelField, validModelName)
 
 	validContents := fmt.Sprintf(`
 a	: b
@@ -240,7 +240,7 @@ foo	: bar
 	data := []testData{
 		{"", "", "", true},
 		{"invalid", "", "", true},
-		{"vendor_id", "", "", true},
+		{archCPUVendorField, "", "", true},
 		{validVendor, "", "", true},
 		{validModel, "", "", true},
 		{validContents, validVendorName, validModelName, false},

--- a/virtcontainers/qemu_arm64.go
+++ b/virtcontainers/qemu_arm64.go
@@ -22,6 +22,9 @@ const defaultQemuMachineType = QemuVirt
 
 const defaultQemuMachineOptions = "gic-version=host,usb=off,accel=kvm"
 
+// Not used
+const defaultPCBridgeBus = ""
+
 var qemuPaths = map[string]string{
 	QemuVirt: defaultQemuPath,
 }
@@ -47,7 +50,7 @@ func MaxQemuVCPUs() uint32 {
 	return uint32(runtime.NumCPU())
 }
 
-func newQemuArch(config HypervisrConfig) qemuArch {
+func newQemuArch(config HypervisorConfig) qemuArch {
 	machineType := config.HypervisorMachineType
 	if machineType == "" {
 		machineType = defaultQemuMachineType

